### PR TITLE
Firat attempt at adding some business logic..

### LINKED ISSUE #6: The application should store and handle student information:
- StudentUser
- Names
-SSN
- Something like physical address
- Email address

### LINKED ISSUE #7: You should be able to create a new student using POST containing basic student info (name, ssn)

### PR DESCRIPTION
**Pull request by simonblund**
resolves #7 
resolves #6 

Take a close look at this request, do we need to keep more information in the student object. The main redesign is that we now keep base information about the student in the microservice for convinience for future development of the project.

### CODE REVIEW

**Comment by weleoka** [comment]
Hey,

Nice work on the implementations. Seems to be sticking to the books of how Spring works. A few thoughts here.

> The main redesign is that we now keep base information about the student in the microservice for convinience for future development of the project.

1) You mean that persistence level is now included in the studentservice... so, how does this play in with the mock-services? Will we guarantee transactions and integrity in the service methods with concurrent API requests to the mock-api and writing to this microservice's persistence level?
All I see now is that this enables the service to be a true microservice... and saves hassles by not having to implement a requests library (and operate on triggers) in student-service.... but sooner or later that has to be done. 

2) Also, I don't aggre that issue #7 belongs here in this service. We have to stick with the task of linking grades to students and course modules vs. ladok. We can't let the project drift and re-write a whole implementation for NyA... or am I missing something?

How do you want to go ahead with this merge? Continue working in the branch or start a new one, because I would vote to not pass this PR as it stands now. I'd like a second opinion @limolback2016 please.

Cheers

**Comment by simonblund** [comment]
I concur with your comment regarding NyA, theoretically we could make that a scheduled task instead. Student-service pulling NyA for new students. I'll have to take a closer look at the mock API before continuing. However for testing purposes an easy to use endpoint to create students is vital.

**Comment by weleoka** [comment]
In fact I think a POST request to mock-services will create a student... it's an in-memory db just manipulated with REST calls and not Repository-classes... which makes me wonder if Spring HTTP requests module/dependency can act like a Repository. Like an ORM around an API.

**Comment by simonblund** [comment]
resolves #9

File: service/build.gradle [hunk 1]
```diff
@@ -9,10 +9,12 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.springframework.cloud:spring-cloud-starter-circuitbreaker-resilience4j'
     implementation 'org.springframework.cloud:spring-cloud-starter-openfeign'
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (weleoka) [inline service/build.gradle:12]: probably drop these dependencies to JPA and h2 database.

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -1 +1,14 @@
...
+    url: jdbc:h2:mem:testdb
+    driver-class-name: org.h2.Driver
+    username: sa
+    password: password
+  jpa:
+    database-platform: org.hibernate.dialect.H2Dialect
+integration:
+  services:
+    mock-service:
+      url: http://localhost:8282
```
Comment (weleoka) [inline service/src/main/resources/application.yml:14]: add a newline at end of file.

**Review by weleoka — APPROVED** [review]
Had a look over this. To get on with the merge I'll approve.

**Comment by weleoka** [comment]
Good work! Made a squash merge to master.

**Comment by weleoka** [comment]
Done. Made the merge.

**Pull request merged**